### PR TITLE
Field-backed properties: add open questions

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

File: proposals/field-keyword.md [hunk 1]
```diff
@@ -598,3 +598,44 @@ partial class C
...
+
+When should the synthesized backing field be considered *read-only*?
+
+```csharp
+struct S
+{
+    object P1          { get { field ??= ""; }}        // ok
+    readonly object P2 { get { field ??= ""; }}        // error: 'field' is readonly
+    readonly object P3 { get; set { _ = field; } }     // ok
+    readonly object P4 { get; set { field = value; } } // error: 'field' is readonly
```
Comment (CyrusNajmabadi) [inline proposals/field-keyword.md:612]: can you add a line where the property has an initializer?

**Review by CyrusNajmabadi — APPROVED** [review]

File: proposals/field-keyword.md (hunk 1 above)
Comment (jnm2) [reply proposals/field-keyword.md:613]: The first line (P0) has one, is that different from what you'd want to see?

**Review by jnm2 — APPROVED** [review]

File: proposals/field-keyword.md (hunk 1 above)
Comment (cston) [reply proposals/field-keyword.md:613]: I added the `P0` example based on the comment from @CyrusNajmabadi.

File: proposals/field-keyword.md [hunk 2]
```diff
@@ -598,3 +598,45 @@ partial class C
...
+            Debug.Assert(field is null);
+            return null;
+        }
+    }
+}
+```
+
+For reference, fields for *primary constructor parameters* are generated in similar cases - see [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRgDoARASwEMBzAOwgGcAXUsOgbgFgAoLjAJhwDCACgjAAVjDAMcAM1IwANgBMAlFwDeXHNrwocAWSFrOOnJpOmdxGMACulQgEE6dGFAZC5ipTlJ0c1LYKCiocFtoAvlxR3JyMULZSOADKIuKS0l7KxuamGHqGxqa5ltrWdg7Oru6e8sq+/oHBoVo6MRFAA).
+
+**Recommendation**: The backing field is generated when `field` is used only in calls to *conditional methods*.
```
Comment (RikkiGibson) [inline proposals/field-keyword.md:642]:
```suggestion
**Recommendation**: The backing field is generated when `field` is used only in omitted calls to *conditional methods*.
```

File: proposals/field-keyword.md (hunk 1 above)
Comment (jnm2) [reply proposals/field-keyword.md:613]: Ah, thanks! Sorry Cyrus!

**Pull request merged**